bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/WebThingHandler.java

```diff
@@ -0,0 +1,301 @@
+            webThingConnectionRef.getAndSet(Optional.empty()).ifPresent(ConsumedThing::close);
+            watchdogHandle.getAndSet(Optional.empty()).ifPresent(future -> future.cancel(true));
+        } finally {
+            super.dispose();
+        }
+    }
+
+    private boolean tryReconnect(@Nullable URI uri) {
+        if (isActivated.get()) { // will try reconnect only, if activated
+            try {
+                // create the client-side WebThing representation
+                if (uri != null) {
+                    var webThing = ConsumedThingFactory.instance().create(webSocketClient, httpClient, uri, scheduler,
+                            this::onError);
+                    this.webThingConnectionRef.getAndSet(Optional.of(webThing)).ifPresent(ConsumedThing::close);
+
+                    // update the Thing structure based on the WebThing description
+                    thingStructureChanged(webThing);
+
+                    // link the Thing's channels with the WebThing properties to forward properties/item updates
+                    establishWebThingChannelLinks(webThing);
+
+                    lastReconnect.set(Instant.now());
+                    updateStatus(ThingStatus.ONLINE);
+                    return true;
+                }
+            } catch (IOException e) {
+                var msg = e.getMessage();
+                if (msg == null) {
+                    msg = "";
+                }
+                onError(msg);
+            }
+        }
+        return false;
+    }
+
+    public void onError(String reason) {
+        var wasConnectedBefore = isOnline();
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, reason);
+
+        // close the WebThing connection. If the handler is still active, the WebThing connection
+        // will be re-established within the periodically watchdog task
+        webThingConnectionRef.getAndSet(Optional.empty()).ifPresent(ConsumedThing::close);
+
+        if (wasConnectedBefore) { // to reduce log messages, just log in case of connection state changed
+            logger.debug("WebThing {} disconnected {}. Try reconnect (each {} sec)", getWebThingLabel(), reason,
+                    HEALTH_CHECK_PERIOD.getSeconds());
+        } else {
+            logger.debug("WebThing {} is offline {}. Try reconnect (each {} sec)", getWebThingLabel(), reason,
+                    HEALTH_CHECK_PERIOD.getSeconds());
+        }
+    }
+
+    private @Nullable URI readWebThingURI() {
+        var webThingConfiguration = getConfigAs(WebThingConfiguration.class);
+        var webThingUriString = webThingConfiguration.webThingURI;
+        if (webThingUriString == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "webThing uri has not been set");
+        } else {
+            try {
+                return URI.create(webThingUriString);
+            } catch (Exception illegalURIException) {
```

Comment:
You should specify the concrete exception you want to catch, since catching `Exception` catches also all runtime exceptions.
```suggestion
            } catch (IllegalArgumentException illegalURIException) {
```